thunder/torch/__init__.py

```diff
@@ -5262,8 +5262,8 @@ def _backward_checkpoint(
 ) -> tuple[None | TensorLike, ...]:
     from thunder.core.transforms import vjp
 
-    result = vjp(function)(args, grad_outputs, **kwargs)
-    return result
+    result, grads = vjp(function)(args, grad_outputs, **kwargs)
+    return grads
```

Comment:
Why did `test_torch_checkpoint` pass?

Reply:
The test passes because the first element from the return tuple (`result` in this code) was registered as a "gradient" for the "function" argument of the checkpoint and simply discarded.

Reply:
Updated the test in https://github.com/Lightning-AI/lightning-thunder/pull/1391.